### PR TITLE
feat: [GEN-2610] Confirmation page redesign

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
- Colocate confirmation page components, hooks, and utils
- Support readOnly mode in CartItem
- Use new CartItem in the order confirmation screen

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4822?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4822** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4822?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4821** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4821?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4812** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4812?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#4813](https://github.com/HedvigInsurance/racoon/pull/4813) <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4813?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#4815](https://github.com/HedvigInsurance/racoon/pull/4815) <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4815?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#4799** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4799?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4796** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4796?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4795** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4795?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4794** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4794?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4793** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4793?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4787** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4788** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4784** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4784?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#4785](https://github.com/HedvigInsurance/racoon/pull/4785) <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4785?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#4778** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4778?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#4786](https://github.com/HedvigInsurance/racoon/pull/4786) <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4786?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#4777** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4777?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4776** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4776?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4775** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4775?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4774** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4774?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4773** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4773?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4772** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4772?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      <main>
+        <div className={container}>
+          <section className={yStack({ gap: { _: 'md', sm: 'lg' } })}>
+            {story ? (
+              <header>
+                <Heading as="h2" variant="standard.24">
+                  {story.content.title}
+                </Heading>
+                <Text color="textSecondary" size="xl">
+                  We’ve sent a copy to your e-mail
```
Comment (alebedev) [inline apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx:34]: i18n?

File: apps/store/src/features/priceCalculator/OfferPresenterV2.tsx [hunk 2]
```diff
@@ -25,6 +24,7 @@ import {
 import { useTiersAndDeductibles } from '@/components/ProductPage/PurchaseForm/useTiersAndDeductibles'
 import { useCartEntryToReplace } from '@/components/ProductPage/useCartEntryToReplace'
 import { DiscountFieldContainer } from '@/components/ShopBreakdown/DiscountFieldContainer'
+import { useOfferDetails } from '@/features/CartItem/hooks/useOfferDetails'
```
Comment (alebedev) [inline apps/store/src/features/priceCalculator/OfferPresenterV2.tsx:27]: Do we need separate subfolder for hooks? They look fine next to the components

File: apps/store/src/features/CartItem/CartItem.tsx [hunk 3]
```diff
@@ -47,15 +48,18 @@ export const CartItem = ({ offer, defaultExpanded }: PropsWithChildren<Props>) =
...
-      <RemoveCartItemDialog.Root shopSessionId={shopSession.id} offer={offer}>
-        <Card.Aside>
-          <RemoveCartItemDialog.Trigger>
-            <IconButton variant="secondary">
-              <CrossIcon />
-            </IconButton>
-          </RemoveCartItemDialog.Trigger>
-        </Card.Aside>
-      </RemoveCartItemDialog.Root>
+      {readOnly ? null : (
```
Comment (alebedev) [inline apps/store/src/features/CartItem/CartItem.tsx:51]: I'm generally in favor of boolean statements vs ternaries in JSX, but here's it's even more important. Compare
```
// 1
readonly ? null : .(...)
// 2
!readonly && (...)
```

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx [hunk 4]
```diff
@@ -24,48 +24,46 @@ export const StaticContent = (props: Props) => {
...
-                          alt={t('APP_DOWNLOAD_QRCODE_ALT')}
-                          width={168}
-                          height={168}
-                          priority={true}
-                        />
-                      </SpaceFlex>
+          <CheckList>
+            {checklistItems.map((item, index) => {
+              const isLast = index === checklistItems.length - 1
+              return isLast ? (
```
Comment (alebedev) [inline apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx:41]: I know this is old code, but can we try to improve the structure?  There's no need to handle last element inside the loop
```
          <CheckList>
            {checklistItems.map((item) => ...}
            {everythingAfterItems}
          </Checklist>
```

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx [hunk 5]
```diff
@@ -24,48 +24,46 @@ export const StaticContent = (props: Props) => {
 
   return (
     <Space y={4}>
```
Comment (alebedev) [inline apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx:26]: Let's get rid of `<Space>` while we're changing it

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+        {switching && <SwitchingAssistantSection {...switching} />}
+
+        {memberPartnerData?.sas?.eligible && (
+          <SasEurobonusSectionContainer
+            initialValue={memberPartnerData.sas.eurobonusNumber ?? ''}
+          />
+        )}
+
+        {/* Treating 'story' as optional here to provide a fallback confirmation page instead crashing the page */}
```
Comment (alebedev) [inline apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx:57]: Nice!

**Review by alebedev — APPROVED** [review]

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx (hunk 1 above)
Comment (Youakeem) [reply apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx:34]: The title comes from Storyblok so we thought to add another field for the subtitle but I am not sure how to do that, can you help?

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx (hunk 4 above)
Comment (Youakeem) [reply apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx:41]: Next PR improve this part but we still use isLast to uncheck the last item

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx (hunk 5 above)
Comment (Youakeem) [reply apps/store/src/app/[locale]/confirmation/[shopSessionId]/components/StaticContent/StaticContent.tsx:26]: Next PR refactor this section, let's address these points there.

File: apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx (hunk 6 above)
Comment (Youakeem) [reply apps/store/src/app/[locale]/confirmation/[shopSessionId]/ConfirmationPage.tsx:57]: That comment was already there so nice job to whoever left it :D

File: apps/store/src/features/CartItem/CartItem.tsx (hunk 3 above)
Comment (Youakeem) [reply apps/store/src/features/CartItem/CartItem.tsx:51]: I understand but I believe ternaries are a better approach to conditional rendering. Is this a hard opinion or can we discuss in frontend forum and agree on a style?

Comment (alebedev) [reply apps/store/src/features/CartItem/CartItem.tsx:51]: Let's discuss on frontend forum

**Comment by Youakeem** [comment]
### Merge activity

* **Oct 8, 3:30 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4822).

**Pull request merged**